.github/workflows/ci.yaml

```diff
@@ -0,0 +1,55 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  schedule:
+    # Run every Wednesday.
+    - cron: '0 0 * * 3'
+
+name: CI
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - run: cargo check --workspace
+
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - run: cargo clippy -- -D warnings
+
+  doc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - run: cargo doc
+        env:
+          RUSTDOCFLAGS: -Dwarnings
```

Comment:
nit: You're using this rust-cache thing, but from its docs it looks to me like the cache won't be shared between these three jobs, which may or may not be intentional?

Setting a shared-key seems like it could fix that, or could just ignore.

Reply:
That's a good point, I've always done it this way but there could be value to sharing the cache. I'll make a todo